scheduler/src/cook/kubernetes/api.clj

```diff
@@ -1334,15 +1334,32 @@
           ; * A job may have additional containers with the name aux-*
           job-status (first (filter (fn [c] (= cook-container-name-for-job (.getName c)))
                                     container-statuses))
-          pod-preempted-timestamp (some-> pod .getMetadata .getLabels (get (or (-> (config/kubernetes) :node-preempted-label) "node-preempted")))
+          pod-metadata (some-> pod .getMetadata)
+          {:keys [node-preempted-label pod-deletion-timeout-seconds]} (config/kubernetes)
+          pod-preempted-timestamp
+          (some-> pod-metadata
+                  .getLabels
+                  (get (or node-preempted-label "node-preempted")))
+          pod-deletion-timestamp (some-> pod-metadata .getDeletionTimestamp)
           synthesized-pod-state
-          (if (some-> pod .getMetadata .getDeletionTimestamp)
-            ; If a pod has been ordered deleted, treat it as if it was gone, It's being async removed.
-            ; Note that we distinguish between this explicit :missing, and not being there at all when processing
-            ; (:cook-expected-state/killed, :missing) in cook.kubernetes.controller/process
-            {:state :missing
-             :reason "Pod was explicitly deleted"
-             :pod-deleted? true}
+          (if pod-deletion-timestamp
+            ; If a pod has been ordered deleted, treat it as if it were gone, unless too
+            ; long (pod-deletion-timeout-seconds) has passed since its deletion timestamp
```

Comment:
Don't like this. Basically we're 'un-:missing' a node if it was deleted too long ago, tunneling a weird state down to the :pod/unknown state, with a special flag for special behavior.  

E.g., this transition:

:pod/running -> :missing -> :pod/unknown (with :hard-delete?) -> :missing !?!

That's really violating the nature of s state machine. We should make a new state :pod/deleting, and map :deletionTimestamp being set to it. Make a state and it makes sense:

:pod/running -> :pod/deleting -> :missing

The state scanner will visit all pods, check the timestamp then and do a hard delete if old enough.

Reply:
Also, unit test.